### PR TITLE
[minor] Add support mongo 5.0.23 and 6.0.12

### PR DESCRIPTION
**Pull request by sanju7216**
- Add mongo community servers version 5.0.23 and 6.0.12.
- Update mongodb kubernetes images to recent ones with security fixes.
- Calculate mongo_feature_compatibility_matrix based on the values from catalog rather than hardcoded.

JIRA task - https://jsw.ibm.com/browse/MASCORE-1173

Default Mongo version from December 2023 catalog will be 5.0.23 with latest mongodb kubernetes images with security fixes.

it will be possible to install specific mongo version setting `MONGODB_VERSION=5.0.23` or `MONGODB_VERSION=6.0.12`.

### CODE REVIEW

File: ibm/mas_devops/roles/mongodb/tasks/providers/community/check-mongo-exists.yml [hunk 1]
```diff
@@ -38,10 +38,30 @@
...
+        file: "{{ role_path }}/../../common_vars/casebundles/{{ catalog_tag }}.yml"
+      when: stat_result is defined and stat_result.stat.exists
+
     # holds the expected target mongo version that is feature compatible with the existing mongo instance version
+    # mongo extras version are picked up from the catalog if available or uses defaults.
     - name: Set mongo_compatible_target_version
       set_fact:
         mongo_compatible_target_version: "{{ mongo_feature_compatibility_matrix[existing_mongo_minor_version] }}"
+      vars:
+        mongo_feature_compatibility_matrix:
```
Comment (andrercm) [inline ibm/mas_devops/roles/mongodb/tasks/providers/community/check-mongo-exists.yml:60]: I think having this var in the `defaults/main.yml` would be easier to find out an maintain going forward, no? just a thought...

Also, do we need to set and maintain a default value here for `mongo_extras_version_X` ? We already define the default for each version in the corresponding casebundle file, so we'll always have a value coming from there, no?

**Review by andrercm — CHANGES_REQUESTED** [review]
Just some food for thought

File: ibm/mas_devops/roles/mongodb/README.md [hunk 2]
```diff
@@ -73,7 +73,7 @@ The namespace where the operator and MongoDb cluster will be deployed.
 - Default Value: `mongoce`
 
 ### mongodb_version
-Defines the specific mongo version to be used. Best practice would be to use the version associated with the current Maximo Application Suite catalog. However, this value can currently be overridden to 4.4.21, 5.0.21 or 6.0.10.
+Defines the specific mongo version to be used. Best practice would be to use the version associated with the current Maximo Application Suite catalog. However, this value can currently be overridden to 4.4.21, 5.0.21, 5.0.23, 6.0.10 oe 6.0.12
```
Comment (terenceq) [inline ibm/mas_devops/roles/mongodb/README.md:76]: small typo here

File: ibm/mas_devops/roles/mirror_extras_prepare/vars/mongoce_6.0.12.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+---
+extra_images:
+  - name: mongodb/mongodb-kubernetes-operator
+    registry: quay.io
+    tag: 0.8.3
+    digest: sha256:9ef7a689b7f2789b436458a6d5f361ffcb4e182daa6c33d79687e87f268c7761 # 0.8.3-b20231215T000000Z
```
Comment (terenceq) [inline ibm/mas_devops/roles/mirror_extras_prepare/vars/mongoce_6.0.12.yml:6]: is it confirmed these images do not disappear from day to day?

File: ibm/mas_devops/roles/mongodb/tasks/providers/community/check-mongo-exists.yml (hunk 1 above)
Comment (sanju7216) [reply ibm/mas_devops/roles/mongodb/tasks/providers/community/check-mongo-exists.yml:60]: @andrercm am using the mongo_extras_version_X value from the case bundle vars..

File: ibm/mas_devops/roles/mirror_extras_prepare/vars/mongoce_6.0.12.yml (hunk 3 above)
Comment (sanju7216) [reply ibm/mas_devops/roles/mirror_extras_prepare/vars/mongoce_6.0.12.yml:6]: These images are tagged with timestamp. So they won't disappear

File: ibm/mas_devops/roles/mongodb/README.md (hunk 2 above)
Comment (sanju7216) [reply ibm/mas_devops/roles/mongodb/README.md:76]: Great spot.. will fix it

File: ibm/mas_devops/roles/mongodb/tasks/providers/community/check-mongo-exists.yml (hunk 1 above)
Comment (sanju7216) [reply ibm/mas_devops/roles/mongodb/tasks/providers/community/check-mongo-exists.yml:60]: I ll move the defaults back to defaults/main.yml

**Review by durera — APPROVED** [review]

**Pull request merged**